### ISSUE TITLE
slf4j error in Asset Management Client (regression after pom refactoring)

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#1228]
After refactoring poms in #1114 and related pom updates, an slf4j runtime error is seen when running the asset management sample (and others).

In the example below a pom change has also been made to build a complete jar with dependencies, which assures a more defined environment than running in IDE
```
➜  asset-management-samples git:(fixslfdep) ✗ java -jar target/asset-management-samples-1.1-SNAPSHOT-jar-with-dependencies.jar fred
===============================
CSV File Reader Sample
===============================
Running against server: cocoMDS1 at http://localhost:8080
Using userId: garygeeke

SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
The connector can not be created from metadata.  Error message is: OMAG-PLATFORM-404-001 The OMAG Server cocoMDS1 is not available to service a request from user garygeeke
===============================
Accessing file: fred
The connector is unable to retrieve the requested record because the file is too short.
```

This was caused when the slf4j-api dependency was removed from 'compile' scope - the mvn dependency:analyze goal does not deem that component as being referenced in the java object code

**Comment by planetf1** [comment]
This error is produced when no slf4j binding can be found on the runtime classpath

Docs at https://www.slf4j.org/manual.html explain in more detail the operation of slf4j. 

One para of note
```
As of SLF4J version 1.6.0, if no binding is found on the class path, then slf4j-api will default to a no-operation implementation discarding all log requests. Thus, instead of throwing a NoClassDefFoundError because the org.slf4j.impl.StaticLoggerBinder class is missing, SLF4J version
1.6.0 and later will emit a single warning message about the absence of a binding and proceed to discard all log requests without further protest.
```
We are prior to 1.6...

It further goes on to provide a recommendation
```
BASIC RULE Embedded components such as libraries or frameworks should not declare a dependency on any SLF4J binding but only depend on slf4j-api. When a library declares a transitive dependency on a specific binding, that binding is imposed on the end-user negating the purpose of SLF4J. Note that declaring a non-transitive dependency on a binding, for example for testing, does not affect the end-user.
```

This appears to suggest that
 * One should NOT include a binding when building components that may be reused
 * Having the same behaviour as no-op -- but without the error -- is CORRECT behaviour
 * In a deployment, a specific binding should be added (at deployment time) if logging is required.

So far... this I think means the dependencies we NOW have IN GENERAL are correct - though we should update slf4j. In fact no components should have a compile dependency on anything other than slf4j-api
 - at least for a base component

It ALSO means that when we create our packaging, we should consider dynamically inserting an appropriate implementation. And one could argue since this particular case is in fact a sample, we should perhaps include a default logging implementation - but that is not through a dependency on slf4j-api, but rather a binding to one of the logging frameworks such as logback (used in many places in egeria already - maybe too many.. ), log4j etc.

**Comment by planetf1** [comment]
Checking our existing poms, they do seem to properly depend on slf4j-api

server-chassis & user-interface both depend on an slf4j implementation -- logback.
This makes sense as they are applications and need to actually control the logging 

So I think the correct approach is to include logback for each of these samples - in fact any standalone tools. This adheres the closest to the spirit of the slf4j docs.

At the same time building up the jar-with-dependencies for these tools makes them easier and consistent to use, so I propose to do this too.

cc: @mandy-chessell

**Comment by planetf1** [comment]
Initial change made to all clients ie:
```
➜  asset-management-samples git:(fixslfdep) ✗ java -jar target/asset-management-samples-1.1-SNAPSHOT-jar-with-dependencies.jar
Please specify the file name in the first parameter
➜  asset-management-samples git:(fixslfdep) ✗ java -jar target/asset-management-samples-1.1-SNAPSHOT-jar-with-dependencies.jar fred
===============================
CSV File Reader Sample
===============================
Running against server: cocoMDS1 at http://localhost:8080
Using userId: garygeeke

08:43:45.902 [main] DEBUG org.odpi.openmetadata.frameworks.connectors.ConnectorProviderBase - Connector class name set: org.odpi.openmetadata.adapters.connectors.restclients.RESTClientConnector
08:43:45.908 [main] DEBUG org.odpi.openmetadata.frameworks.connectors.ConnectorProviderBase - Connector class name set: org.odpi.openmetadata.adapters.connectors.restclients.spring.SpringRESTClientConnector
```

Observations
 1. logback classic uses DEBUG as the default log level. I think that is too verbose/unintended for the sample use case -> can keep with logback (as we use elsewhere) and add property to set level higher such as INFO
2. Note that the built jar has been named after the project. this feels too verbose and cumbersome. However it is consistent, so will leave as is for now. See open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/pom.xml for an example in renaming the executable jar
3. We should probably consider adding samples into a proper assembly - and may further consider renames at that point. I will take this as out of scope for this particular issue

**Comment by planetf1** [comment]
Having made this change the output now looks like:
```
➜  asset-management-samples git:(fixslfdep) ✗ java -jar target/asset-management-samples-1.1-SNAPSHOT-jar-with-dependencies.jar fred
===============================
CSV File Reader Sample
===============================
Running against server: cocoMDS1 at http://localhost:8080
Using userId: garygeeke

The connector can not be created from metadata.  Error message is: OMAG-PLATFORM-404-001 The OMAG Server cocoMDS1 is not available to service a request from user garygeeke
===============================
Accessing file: fred
The connector is unable to retrieve the requested record because the file is too short.
```
This is with a logback.xml specifying log level of info. Of course there are none, so we see no debug, but only the regular stdout. This begs the question as to whether in fact omitting logback, and instead upgrading slf to >1.6, and allowing the default behaviour (no logging) is in fact the better, simpler option.

This should also be considered for chassis, UI backend code - as it makes changing the logging implementation in a deployment potentially easier (for example perhaps the base code should not use a logger, but we add it during docker deployment). Will leave for now.

**Comment by planetf1** [comment]
slf4f is already 1.7.25, so surprised we see the initial error.

An alternative option is to include slf4j-simple (logs everything including debug), or slf4j-nop (discards all debug logging)- following the same pattern as logback, but with no requirement for an additional resource file.

Given though we're already using logback,and it makes it easier to override (adding new xml in classpath) I will use that approach for now. Please comment in the PR if you'd prefer one of the other approaches.

**Comment by planetf1** [comment]
A full example of asset management:
```
➜  open-metadata-samples git:(fixslfdep) ✗ java -jar access-services-samples/asset-management-samples/target/asset-management-samples-1.1-SNAPSHOT-jar-with-dependencies.jar access-services-samples/asset-management-samples/ContactList.csv
===============================
CSV File Reader Sample
===============================
Running against server: cocoMDS1 at http://localhost:8080
Using userId: garygeeke

The connector can not be created from metadata.  Error message is: REPOSITORY-HANDLER-400-003 An unexpected error was returned by the property server during getAssetsByName request for open metadata access service Asset Consumer OMAS on server cocoMDS1; message was OMRS-REPOSITORY-404-007 The TypeDef unique identifier 896d14c2-7522-4f6c-8519-757711943fe6 passed as parameter entityTypeGUID on a findEntitiesByProperty request to open metadata repository cocoMDS1 is not known to this repository
===============================
Accessing file: access-services-samples/asset-management-samples/ContactList.csv
Number of records: 26
First 10 records ...
------------------------------------------------------------------------
[RecId, ContactType, FirstName, LastName, Company, JobTitle, WorkLocation]
------------------------------------------------------------------------
[1, E, Zach, Now, Coco Pharmaceuticals, Founder, 3]
[2, E, Steve, Starter, Coco Pharmaceuticals, Founder, 1]
[3, E, Terri, Daring, Coco Pharmaceuticals, Founder, 2]
[4, E, Tanya, Tidy, Coco Pharmaceuticals, "Data Steward, Clinical Trials, 3]
[5, E, Polly, Tasker, Coco Pharmaceuticals, IT Project Leader, 1]
[6, E, Tessa, Tube, Coco Pharmaceuticals, "Lead Researcher, Clinical Trials, 3]
[7, E, Callie, Quartile, Coco Pharmaceuticals, Data Scientist, 3]
[8, E, Ivor, Padlock, Coco Pharmaceuticals, Chief Security Officer, 1]
[9, E, Bob, Nitter, Coco Pharmaceuticals, Application Developer, 1]
[10, E, Faith, Broker, Coco Pharmaceuticals, Human Resources and Compliance Director, 1]
------------------------------------------------------------------------
No asset properties  ...
➜  open-metadata-samples git:(fixslfdep) ✗
```
The interaction with the repo is failing

repeating the same with in-memory-repository INSTEAD of the graph results in a different error
```
➜  open-metadata-samples git:(fixslfdep) ✗ java -jar access-services-samples/asset-management-samples/target/asset-management-samples-1.1-SNAPSHOT-jar-with-dependencies.jar access-services-samples/asset-management-samples/ContactList.csv
===============================
CSV File Reader Sample
===============================
Running against server: cocoMDS1 at http://localhost:8080
Using userId: garygeeke

The connector can not be created from metadata.  Error message is: REPOSITORY-HANDLER-400-003 An unexpected error was returned by the property server during getAssetsByName request for open metadata access service Asset Consumer OMAS on server cocoMDS1; message was OMRS-ENTERPRISE-REPOSITORY-503-001 There are no open metadata repositories available for access service Asset Consumer.
===============================
Accessing file: access-services-samples/asset-management-samples/ContactList.csv
Number of records: 26
First 10 records ...
------------------------------------------------------------------------
[RecId, ContactType, FirstName, LastName, Company, JobTitle, WorkLocation]
------------------------------------------------------------------------
[1, E, Zach, Now, Coco Pharmaceuticals, Founder, 3]
[2, E, Steve, Starter, Coco Pharmaceuticals, Founder, 1]
[3, E, Terri, Daring, Coco Pharmaceuticals, Founder, 2]
[4, E, Tanya, Tidy, Coco Pharmaceuticals, "Data Steward, Clinical Trials, 3]
[5, E, Polly, Tasker, Coco Pharmaceuticals, IT Project Leader, 1]
[6, E, Tessa, Tube, Coco Pharmaceuticals, "Lead Researcher, Clinical Trials, 3]
[7, E, Callie, Quartile, Coco Pharmaceuticals, Data Scientist, 3]
[8, E, Ivor, Padlock, Coco Pharmaceuticals, Chief Security Officer, 1]
[9, E, Bob, Nitter, Coco Pharmaceuticals, Application Developer, 1]
[10, E, Faith, Broker, Coco Pharmaceuticals, Human Resources and Compliance Director, 1]
------------------------------------------------------------------------
No asset properties  ...
```
This output is as we expect

So this error this isn't directly related to logging. Therefore will complete this fix/raise PR and then raise new PR if there are functional issues after more investigation

**Comment by planetf1** [comment]
After retesting with cleanly build server no errors were displayed ie
```
➜  asset-management-samples git:(springupd) java -jar target/asset-management-samples-1.1-SNAPSHOT-jar-with-dependencies.jar ContactList.csv
===============================
CSV File Reader Sample
===============================
Running against server: cocoMDS1 at http://localhost:8080
Using userId: garygeeke

The open metadata repositories do not have an asset definition for the requested file name ContactList.csv
===============================
Accessing file: ContactList.csv
Number of records: 26
First 10 records ...
------------------------------------------------------------------------
[RecId, ContactType, FirstName, LastName, Company, JobTitle, WorkLocation]
------------------------------------------------------------------------
[1, E, Zach, Now, Coco Pharmaceuticals, Founder, 3]
[2, E, Steve, Starter, Coco Pharmaceuticals, Founder, 1]
[3, E, Terri, Daring, Coco Pharmaceuticals, Founder, 2]
[4, E, Tanya, Tidy, Coco Pharmaceuticals, "Data Steward, Clinical Trials, 3]
[5, E, Polly, Tasker, Coco Pharmaceuticals, IT Project Leader, 1]
[6, E, Tessa, Tube, Coco Pharmaceuticals, "Lead Researcher, Clinical Trials, 3]
[7, E, Callie, Quartile, Coco Pharmaceuticals, Data Scientist, 3]
[8, E, Ivor, Padlock, Coco Pharmaceuticals, Chief Security Officer, 1]
[9, E, Bob, Nitter, Coco Pharmaceuticals, Application Developer, 1]
[10, E, Faith, Broker, Coco Pharmaceuticals, Human Resources and Compliance Director, 1]
------------------------------------------------------------------------
No asset properties  ...
```
